### PR TITLE
Add email validation for coupon form.

### PR DESCRIPTION
**Pull request by vkaracic**
It was possible to add anything in the email domain input field. This PR contains pattern validation for the email domain field.

https://openedx.atlassian.net/browse/SOL-2034

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=955&repo=edx%2Fecommerce

**Review by iivic — APPROVED** [review]
A---....a.bf
This is a valid domain and the only questionable case I could find. If it's acceptable I approve.

File: ecommerce/static/js/models/coupon_model.js [hunk 1]
```diff
@@ -108,7 +108,10 @@ define([
                         return Backbone.Validation.messages.seat_types;
                     }
                 },
-                email_domains: {required: false},
+                email_domains: {
+                    pattern: /^[\w\d][\w\d-\.]*[\w\d]\.[a-z]{2,11}(,[\w\d][\w\d-\.]*[\w\d]\.[a-z]{2,11})*$/,
```
Comment (marjev) [inline ecommerce/static/js/models/coupon_model.js]: what about special characters in the domain name? Should domain name start with a number?

Check this: http://regexlib.com/Search.aspx?k=email&c=-1&m=5&ps=20

Comment (vkaracic) [reply ecommerce/static/js/models/coupon_model.js]: Domain names can only have dots and hyphens as special characters.
Domains can start with a number. https://www.000webhost.com/

Comment (vkaracic) [reply ecommerce/static/js/models/coupon_model.js]: I ended up using the one from here: https://www.safaribooksonline.com/library/view/regular-expressions-cookbook/9781449327453/ch08s15.html

**Review by marjev — APPROVED** [review]

**Pull request merged**